src/modules/battery_status/battery_status.cpp

```diff
@@ -190,33 +183,43 @@ BatteryStatus::adc_poll()
 
 	int selected_source = -1;
 
-	if (ret >= (int)sizeof(buf_adc[0])) {
-
-		/* Read add channels we got */
-		for (unsigned i = 0; i < ret / sizeof(buf_adc[0]); i++) {
-			{
-				for (int b = 0; b < BOARD_NUMBER_BRICKS; b++) {
-
-					/* Once we have subscriptions, Do this once for the lowest (highest priority
-					 * supply on power controller) that is valid.
-					 */
-					if (selected_source < 0 && _analogBatteries[b]->is_valid()) {
-						/* Indicate the lowest brick (highest priority supply on power controller)
-						 * that is valid as the one that is the selected source for the
-						 * VDD_5V_IN
-						 */
-						selected_source = b;
-
-					}
-
-					/* look for specific channels and process the raw voltage to measurement data */
-					if (_analogBatteries[b]->get_voltage_channel() == buf_adc[i].am_channel) {
-						/* Voltage in volts */
-						bat_voltage_adc_readings[b] = buf_adc[i].am_data;
-
-					} else if (_analogBatteries[b]->get_current_channel() == buf_adc[i].am_channel) {
-						bat_current_adc_readings[b] = buf_adc[i].am_data;
-					}
+	adc_report_s adc_report;
+
+	if (_adc_report_sub.update(&adc_report)) {
+
```

Comment:
This change look incorrect.

There is 1 set of channels. Of which both bricks values are listed in that set of channels.
It was once through the the list of channels and selected the one brick's values.
Now it it doing multiple passes on the channels

Reply:
Fixed with 63c08cb